tests/extensions/test_custom.py

```diff
@@ -40,35 +48,29 @@ def test_prop(self) -> Optional[str]:
     def test_prop(self, v: Optional[str]) -> None:
         self._set_property(TEST_PROP, v)
 
-    def add_link(self, target: pystac.STACObject) -> None:
-        if self.obj is not None:
-            self.obj.add_link(pystac.Link(TEST_LINK_REL, target))
-        else:
-            raise pystac.ExtensionAlreadyExistsError(f"{self} does not support links")
-
     @classmethod
     def get_schema_uri(cls) -> str:
-        raise NotImplementedError
+        return SCHEMA_URI
 
-    @staticmethod
-    def custom_ext(obj: T) -> "CustomExtension[T]":
+    @classmethod
+    def ext(cls, obj: T, add_if_missing: bool = False) -> "CustomExtension[T]":
         if isinstance(obj, pystac.Asset):
+            cls.validate_owner_has_extension(obj, add_if_missing)
             return cast(CustomExtension[T], AssetCustomExtension(obj))
         if isinstance(obj, pystac.Item):
+            cls.validate_has_extension(obj, add_if_missing)
             return cast(CustomExtension[T], ItemCustomExtension(obj))
         if isinstance(obj, pystac.Collection):
+            cls.validate_has_extension(obj, add_if_missing)
             return cast(CustomExtension[T], CollectionCustomExtension(obj))
         if isinstance(obj, pystac.Catalog):
+            cls.validate_has_extension(obj, add_if_missing)
             return cast(CustomExtension[T], CatalogCustomExtension(obj))
 
         raise pystac.ExtensionTypeError(
             f"Custom extension does not apply to {type(obj).__name__}"
         )
 
-    @staticmethod
-    def summaries(obj: pystac.Collection) -> "SummariesCustomExtension":
-        return SummariesCustomExtension(obj)
-
 
 class CatalogCustomExtension(CustomExtension[pystac.Catalog]):
```

Comment:
`self.catalog = catalog` on L77 has no effect, never accessed that I can see. It is repeated on L84, L91, and L98. Those lines can be removed.